### PR TITLE
Add raw recommendations

### PR DESCRIPTION
**Pull request by amCap1712**
Add a new category of recommendations - raw. These recommendations are directly calculated from the model based on the training data without passing any candidate data set to the model.

### CODE REVIEW

File: listenbrainz_spark/recommendations/recording/recommend.py [hunk 1]
```diff
@@ -87,9 +87,21 @@ def process_recommendations(recommendation_df, limit):
               FROM recommendation
         )
         SELECT user_id
-             , recording_mbid
-             , score
-             , date_format(latest_listened_at, "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'") AS latest_listened_at
+             , array_sort(
```
Comment (amCap1712) [inline listenbrainz_spark/recommendations/recording/recommend.py:90]: This is change is not to do with raw_recs but it felt to me that collecting in a list in python was slower and doing it in spark seemed somewhat faster.

File: listenbrainz_spark/recommendations/recording/recommend.py [hunk 2]
```diff
@@ -170,47 +182,48 @@ def get_user_name_and_user_id(top_artist_candidate_set_df, users):
...
-        user_rec[row.user_id]['similar_artist'].append(
-            {
-                "recording_mbid": row.recording_mbid,
-                "score": row.score,
-                "latest_listened_at": row.latest_listened_at
-            }
-        )
+        row_dict = row.asDict(recursive=True)
+        user_rec[row_dict["user_id"]]["similar_artist"] = row_dict["recs"]
+        similar_artist_rec_user_count = similar_artist_rec_user_count + 1
```
Comment (mayhem) [inline listenbrainz_spark/recommendations/recording/recommend.py:219]: Any reason not to use += here?

**Review by mayhem — APPROVED** [review]
I'm looking forward to seeing these in action!

File: listenbrainz_spark/recommendations/recording/recommend.py (hunk 2 above)
Comment (amCap1712) [reply listenbrainz_spark/recommendations/recording/recommend.py:219]: No reason, will fix.

**Pull request merged**